Module-2-Basic-Programming/Day-1/Tasks/Mujaheed.task

```diff
@@ -0,0 +1,52 @@
+1. Explain about variable hoistiog with reasoning ?
+
+A. 1. Declarations are hoisted, not initializations.
+   2. Allows calling functions before their declarations.
+   3. All variable and function declarations are processed before any code execution.
+   4. Undeclared variables are implicitly created as global variables when assigned a value.
+
+    Example Program:
+
+    function codejs(){
+    a = 10;
+    let b = 50;                           
```

Comment:
from this function the variable b is undefined and where is var initialisation i cannot see where u have declared variable 'a'?